l4project/exercises_system_project/exercises_system_project/rubbish/settings old.txt
# Django settings for exercises_system_project project.

import os
SETTINGS_DIR = os.path.dirname(__file__)
PROJECT_PATH = os.path.join(SETTINGS_DIR, os.pardir)
PROJECT_PATH = os.path.abspath(PROJECT_PATH)
TEMPLATE_PATH = os.path.join(PROJECT_PATH, 'templates')
STATIC_PATH = os.path.join(PROJECT_PATH,'static')

DEBUG = True
TEMPLATE_DEBUG = DEBUG 

ADMINS = (
    # ('Your Name', 'your_email@example.com'),
)

MANAGERS = ADMINS

DATABASE_PATH = os.path.join(PROJECT_PATH, 'exerciser.db')

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3', # Add 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
        'NAME': DATABASE_PATH,                      # Or path to database file if using sqlite3.
        # The following settings are not used with sqlite3:
        'USER': '',
        'PASSWORD': '',
        'HOST': '',                      # Empty for localhost through domain sockets or '127.0.0.1' for localhost through TCP.
        'PORT': '',                      # Set to empty string for default.
    }
}

# Hosts/domain names that are valid for this site; required if DEBUG is False
# See https://docs.djangoproject.com/en/1.5/ref/settings/#allowed-hosts
ALLOWED_HOSTS = []

# Local time zone for this installation. Choices can be found here:
# http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
# although not all choices may be available on all operating systems.
# In a Windows environment this must be set to your system time zone.
TIME_ZONE = 'America/Chicago'

# Language code for this installation. All choices can be found here:
# http://www.i18nguy.com/unicode/language-identifiers.html
LANGUAGE_CODE = 'en-us'

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = True

# If you set this to False, Django will not format dates, numbers and
# calendars according to the current locale.
USE_L10N = True

# If you set this to False, Django will not use timezone-aware datetimes.
USE_TZ = True

# Absolute filesystem path to the directory that will hold user-uploaded files.
# Example: "/var/www/example.com/media/"
MEDIA_ROOT = os.path.join(PROJECT_PATH, 'media') # Absolute path to the media directory

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
# Examples: "http://example.com/media/", "http://media.example.com/"
MEDIA_URL = '/media/'

# Absolute path to the directory static files should be collected to.
# Don't put anything in this directory yourself; store your static files
# in apps' "static/" subdirectories and in STATICFILES_DIRS.
# Example: "/var/www/example.com/static/"
STATIC_ROOT = ''

# URL prefix for static files.
# Example: "http://example.com/static/", "http://static.example.com/"
STATIC_URL = '/static/'

# Additional locations of static files
STATICFILES_DIRS = (
    # Put strings here, like "/home/html/static" or "C:/www/django/static".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
	STATIC_PATH,
)

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
)

# Make this unique, and don't share it with anybody.
SECRET_KEY = 'gkqa=$!806(j62kf*(fi7+(j76f7nyp)crwzh++hq*qu%9fpwg'

# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
#     'django.template.loaders.eggs.Loader',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    # Uncomment the next line for simple clickjacking protection:
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'exercises_system_project.urls'

# Python dotted path to the WSGI application used by Django's runserver.
WSGI_APPLICATION = 'exercises_system_project.wsgi.application'

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    TEMPLATE_PATH
)

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # Uncomment the next line to enable the admin:
    'django.contrib.admin',
    # Uncomment the next line to enable admin documentation:
    # 'django.contrib.admindocs',
    'exerciser'
)

SESSION_SERIALIZER = 'django.contrib.sessions.serializers.JSONSerializer'

# A sample logging configuration. The only tangible logging
# performed by this configuration is to send an email to
# the site admins on every HTTP 500 error when DEBUG=False.
# See http://docs.djangoproject.com/en/dev/topics/logging for
# more details on how to customize your logging configuration.
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'filters': {
        'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse'
        }
    },
    'handlers': {
        'mail_admins': {
            'level': 'ERROR',
            'filters': ['require_debug_false'],
            'class': 'django.utils.log.AdminEmailHandler'
        }
    },
    'loggers': {
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': True,
        },
    }
}



#maybe delete if it doesn't work
TEMPLATE_CONTEXT_PROCESSORS = ("django.contrib.auth.context_processors.auth",
	"django.core.context_processors.debug",
	"django.core.context_processors.i18n",
	"django.core.context_processors.media",
	"django.core.context_processors.static",
	"django.contrib.messages.context_processors.messages",
	"exerciser.context_processor.applications")

l4project/exercises_system_projectworksbeforetests/exercises_system_project/static/rubbish/change options almost working.txt
var currentStep = 0; // Stores the current step you're on. 0 means initial state.
var d = new Date();
var lastTime = d.getTime();
var csrftoken = getCookie('csrftoken');
var direction = "next";
var answer = "";
var explanation_dict={}

$("#btn_prev").hide();


function goToStep(direction) {

	var totalSteps = steps.length; // Total number of possible steps
	$("*[id^='fragment_']").css("background-color", "transparent");
	$('#explanation').html("");
	direction = direction;

	if (direction == "back") {
		currentStep--;
	}


	if (currentStep == totalSteps-1 && direction == "next"){
		$("#btn_next").hide();
	}
	else if(currentStep == 0 && direction == "back") {
		$("#btn_prev").hide();
	}
	else {
		$("#btn_prev").show();
		$("#btn_next").show();
	}

	if(currentStep >= 0 && currentStep < totalSteps){	
		for (var i = 0; i < steps[currentStep].length; i++) {
			var currentAction = steps[currentStep][i];
			var text = currentAction[0];
			var action = currentAction[1];
			// If we want to go back, we need to reverse the action!
			if (direction == "back") {
				action = undoMapping[action];
			}
			
			if (action == "question"){
				if(direction == "next" && explanation_dict[currentStep] == undefined){
					var options = currentAction[2];
					askQuestion(text, options);
				}
			}
			else{
				doAction(text, action); // Do the action! Show, hide, and (eventually) highlight/unhighlight.
			}
			
		}
		if(action != "question"){
			var now = new Date().getTime();
			$.post("/exerciser/log_info/",
			{
			time : (now - lastTime) / 1000,
			step : currentStep + 1,
			direction : direction,
			answer : answer,
			csrfmiddlewaretoken : csrftoken
			});
			lastTime = now;
		}

		
		if (direction == "next" && explanation_dict[currentStep] == undefined) {
		
			var explanationText= answer +'Step '+ (currentStep + 1) +"/" + totalSteps + ': <br>';
			if(action!="question"){
				explanationText +=  explanations[currentStep].substring(1, explanations[currentStep].length-1);
				answer = "";
			}
			else {
				explanationText += text;
				for (var option_num = 0; option_num < options.length; option_num++){
					explanationText += "<br>" + options[option_num];
				}
			}
			$('#explanation').html(explanationText);
			explanation_dict[currentStep] = explanationText;
			currentStep++;

		}
		else if (direction == "next" && explanation_dict[currentStep] != undefined){
			$('#explanation').html(explanation_dict[currentStep]);
			currentStep++;

		}
		else {
			$('#explanation').html(explanation_dict[currentStep-1]);
		}
	}

}


// When we go back, we need to undo what we did. So this maps an action to its inverse (e.g. when you revert a show, you hide it).
var undoMapping = {
	'show': 'hide',
	'hide': 'show',
	'highlight' : 'unhighlight',
	'unhighlight' : 'highlight',
	'question' : 'question',
}


function doReset() {
	$("*[id^='fragment_']").css("background-color", "transparent");
	$("*[id^='fragment_']").hide();
	$('#explanation').html('');
	$("#btn_prev").hide();
	currentStep = 0;
}


function doAction(fragment, action) {
	var fragmentId = "fragment_" + fragment;
	var object = $('#' + fragmentId);
	
	if (action == "show") {
		object.show();
		object.css("background-color", "#BC8F8F");
	}
	
	else if (action == "hide") {
		object.hide();
	}
	
	else if (action == "highlight") {
		object.css("background-color", "#DA70D6");
	}
	
	else if (action == "unhighlight") {
		object.css("background-color", "transparent");
	}
}

function askQuestion(questionText, options){
	$('#question_text').text(questionText);
	$("#options").empty();
	for (var option_num = 0; option_num < options.length; option_num++){
		var option = options[option_num];
		option_elem = "<input class = 'option' id='option".concat(option_num,"' name='option' type='radio' value = \"", option, "\"/><label for='option",option_num,"'>",option,"</label><br>");
		$("#options").append(option_elem);
	}
	ShowDialog();
}

// Use JQuery to pick up when the user pushes the next button.
$('#btn_next').click(function() {
	goToStep("next");
});

// And again, bind an event to the previous button.
$('#btn_prev').click(function() {
	goToStep("back");
});

$('#btn_reset').click(function() {
	doReset();
});


$(document).ready(function ()
{
	$("#btnClose").click(function (e){
		HideDialog();
		e.preventDefault();
	});

	$("#btnSubmit").click(function (e){
		answer = $(".options input[type='radio']:checked").val();
		explanation_dict[currentStep-1] = " You answered: " + answer + "<br>" + explanation_dict[currentStep-1];
		HideDialog();
		e.preventDefault();
		var now = new Date().getTime();
		$.post("/exerciser/log_info/",
		{	time : (now - lastTime) / 1000,
			step : currentStep,
			direction : direction,
			answer : " The answer the user selected is: " + answer,
			csrfmiddlewaretoken : csrftoken
		});
		lastTime = now;
		answer = " You answered: " + answer + "<br>";
		goToStep("next");
	});

});

function ShowDialog(){
	$("#overlay").show();
	$("#dialog").fadeIn(300);
	$("#overlay").unbind("click");
}

function HideDialog(){
	$("#overlay").hide();
	$("#dialog").fadeOut(300);
} 

function getCookie(name) {
    var cookieValue = null;
    if (document.cookie && document.cookie != '') {
        var cookies = document.cookie.split(';');
        for (var i = 0; i < cookies.length; i++) {
            var cookie = jQuery.trim(cookies[i]);
            // Does this cookie string begin with the name we want?
            if (cookie.substring(0, name.length + 1) == (name + '=')) {
                cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
                break;
            }
        }
    }
    return cookieValue;
}


l4project/exercises_system_project/exercises_system_project/settings.py
# Django settings for exercises_system_project project.

import os
SETTINGS_DIR = os.path.dirname(__file__)
PROJECT_PATH = os.path.join(SETTINGS_DIR, os.pardir)
PROJECT_PATH = os.path.abspath(PROJECT_PATH)
TEMPLATE_PATH = os.path.join(PROJECT_PATH, 'templates')
STATIC_PATH = os.path.join(PROJECT_PATH,'static')

DEPLOYED = False
DEBUG = True
TEMPLATE_DEBUG = DEBUG

ADMINS = (
    # ('Your Name', 'your_email@example.com'),
)

MANAGERS = ADMINS

DATABASE_PATH = os.path.join(PROJECT_PATH, 'exerciser.db')

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3', # Add 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
        'NAME': DATABASE_PATH,                      # Or path to database file if using sqlite3.
        # The following settings are not used with sqlite3:
        'USER': '',
        'PASSWORD': '',
        'HOST': '',                      # Empty for localhost through domain sockets or '127.0.0.1' for localhost through TCP.
        'PORT': '',                      # Set to empty string for default.
    }
}

# Hosts/domain names that are valid for this site; required if DEBUG is False
# See https://docs.djangoproject.com/en/1.5/ref/settings/#allowed-hosts
ALLOWED_HOSTS = []

# Local time zone for this installation. Choices can be found here:
# http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
# although not all choices may be available on all operating systems.
# In a Windows environment this must be set to your system time zone.
TIME_ZONE = 'America/Chicago'

# Language code for this installation. All choices can be found here:
# http://www.i18nguy.com/unicode/language-identifiers.html
LANGUAGE_CODE = 'en-us'

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = True

# If you set this to False, Django will not format dates, numbers and
# calendars according to the current locale.
USE_L10N = True

# If you set this to False, Django will not use timezone-aware datetimes.
USE_TZ = True

# Absolute filesystem path to the directory that will hold user-uploaded files.
# Example: "/var/www/example.com/media/"
MEDIA_ROOT = os.path.join(PROJECT_PATH, 'media') # Absolute path to the media directory

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
# Examples: "http://example.com/media/", "http://media.example.com/"
MEDIA_URL = '/media/'

# Absolute path to the directory static files should be collected to.
# Don't put anything in this directory yourself; store your static files
# in apps' "static/" subdirectories and in STATICFILES_DIRS.
# Example: "/var/www/example.com/static/"
STATIC_ROOT = ''

# URL prefix for static files.
# Example: "http://example.com/static/", "http://static.example.com/"
if DEPLOYED:
    STATIC_URL = '/weave/static/'
else:
    STATIC_URL = '/static/'

# Additional locations of static files
STATICFILES_DIRS = (
    # Put strings here, like "/home/html/static" or "C:/www/django/static".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
	STATIC_PATH,
)

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
)

# Make this unique, and don't share it with anybody.
SECRET_KEY = 'gkqa=$!806(j62kf*(fi7+(j76f7nyp)crwzh++hq*qu%9fpwg'

# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
#     'django.template.loaders.eggs.Loader',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'exerciser.disable.DisableCSRF',
    # Uncomment the next line for simple clickjacking protection:
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'exercises_system_project.urls'

# Python dotted path to the WSGI application used by Django's runserver.
WSGI_APPLICATION = 'exercises_system_project.wsgi.application'

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    TEMPLATE_PATH
)

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # Uncomment the next line to enable the admin:
    'django.contrib.admin',
    # Uncomment the next line to enable admin documentation:
    # 'django.contrib.admindocs',
    'exerciser',
	'chartit'
)

LOGIN_URL = '/exerciser/login/'

SESSION_SERIALIZER = 'django.contrib.sessions.serializers.JSONSerializer'

# A sample logging configuration. The only tangible logging
# performed by this configuration is to send an email to
# the site admins on every HTTP 500 error when DEBUG=False.
# See http://docs.djangoproject.com/en/dev/topics/logging for
# more details on how to customize your logging configuration.

if DEPLOYED:
    log_filename = '/var/www/DJANGO/weave/mysite/weave.log'
else:
    log_filename = 'C:\Users\Emi\mysite.log'

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'filters': {
        'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse'
        }
    },
	'formatters': {
        'verbose': {
            'format' : "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
            'datefmt' : "%d/%b/%Y %H:%M:%S"
        },
        'simple': {
            'format': '%(levelname)s %(message)s'
        },
    },
    'handlers': {
        'mail_admins': {
            'level': 'ERROR',
            'filters': ['require_debug_false'],
            'class': 'django.utils.log.AdminEmailHandler'
        },
        'file': {
            'level': 'DEBUG',
            'class': 'logging.FileHandler',
            'filename': log_filename,
            'formatter': 'verbose'
        },
    },
    'loggers': {
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': True,
        },
        'exerciser': {
            'handlers': ['file'],
            'level': 'DEBUG',
        },
    }
}



#maybe delete if it doesn't work
TEMPLATE_CONTEXT_PROCESSORS = ("django.contrib.auth.context_processors.auth",
	"django.core.context_processors.debug",
	"django.core.context_processors.i18n",
	"django.core.context_processors.media",
	"django.core.context_processors.static",
	"django.contrib.messages.context_processors.messages",
	"exerciser.context_processor.applications",
	"django.core.context_processors.request")
